### PR TITLE
Feature/cross

### PR DESCRIPTION
**Pull request by lasote**
- Pending review docs about the use of system autodetection instead of os_build and arch_build
- Pending to explain the behavior of the build settings in the package id and how to change it

### CODE REVIEW

File: systems_cross_building/windows_subsystems.rst [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    - **msys2_installer/latest@bincrafters/stable**: MSYS2 subsystem as a Conan package.
+    - **cygwin_installer/2.9.0@bincrafters/stable**: Cygwin subsystem as a Conan package.
+
+- From Conan-center:
+
+    - **mingw_installer/1.0@conan/stable**: MinGW compiler installer as a Conan package.
+
+- From Conan-community: ``conan remote add community https://api.bintray.com/conan/conan-community/conan``
+
+    - **vs_path_wrapper/1.0@conan/stable**: Wrapper of your own Visual Studio compiler, to be able to put it
```
Comment (memsharded) [inline systems_cross_building/windows_subsystems.rst]: I still don't see why this deserves a build require, cannot be made with a ``tool.vs_path_order()`` helper? Cannot be coded in the recipe to be self-contained and do not require an external build-require?

File: reference/build_helpers/autotools.rst [hunk 2]
```diff
@@ -40,69 +40,63 @@ It also works using the :ref:`environment_append <environment_append_tool>` cont
...
 .. code-block:: python
 
     class AutoToolsBuildEnvironment(object):
 
-        def __init__(self, conanfile)
+        def __init__(self, conanfile, win_bash=False)
 
 Parameters:
     - **conanfile** (Required): Conanfile object. Usually ``self`` in a conanfile.py
+    - **win_bash**: Defaulted to False. When True, it will run the configure/make commands inside a
```
Comment (danimtb) [inline reference/build_helpers/autotools.rst]: Follow convention for parameter description (this important for the reference section):
- **win_bash** (Optional, Defaulted to ``False``): here the explanation

File: devtools/build_requires.rst [hunk 3]
```diff
@@ -46,7 +46,7 @@ Build requirements can be also specified in a package recipe, with the ``build_r
 
         def build_requirements(self):
             # useful for example for conditional build_requires
-            if self.settings.os == "Windows":
+            if self.settings.os_build == "Windows":
```
[the remark this answers is not in the document]
Comment (lasote) [reply devtools/build_requires.rst]: Mhh, but being build_requires, by definition, will be run on the `build` system, not in the `host` system. So shouldn't be the testing framework a regular ``require`` to be packaged together with the library? is the only way to run the tests in the host.

File: devtools/create_installer_packages.rst [hunk 4]
```diff
@@ -4,125 +4,127 @@
...
 ============================================
 
+Conan 1.0 introduced two new settings, ``os_build`` and ``arch_build``. These settings represent
+the machine where Conan is running, and are important settings when we are packaging tools.
+
+These settings are different from ``os`` and ``arch``. These mean where the built software by the Conan
+recipe will run. When we are packaging a tool, it usually makes no sense, because we are not
+building any software, but it makes sense if you are :ref:`cross building software<cross_building>`.
+
+We recommend the use of ``os_build`` and ``arch_build`` settings instead of ``os`` and ``arch`` if you are
```
[the remark this answers is not in the document]
Comment (lasote) [reply devtools/create_installer_packages.rst]: I get your idea, but the section is called "Creating conan packages to install dev tools", so we are again, specifying that we will run these requirements in the building process? So always build machine?

File: devtools/build_requires.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (memsharded) [reply devtools/build_requires.rst]: Yes, I agree it is edge-case, maybe automating the delivery of the built tests to some external system that will run them (like doing and sftp, ssh or something)

File: devtools/create_installer_packages.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (memsharded) [reply devtools/create_installer_packages.rst]: Yes, but the concept of what we call dev-tools, and how they are intended to be used by conan has not been introduced yet. So explictly putting some examples, like cmake, mingw, makes sense to me.

File: systems_cross_building/windows_subsystems.rst (hunk 1 above)
Comment (lasote) [reply systems_cross_building/windows_subsystems.rst]: I will try to modify the current `tool.vcvars` context manager, but the problem is, the run_in_windows_bash function reads the `conanfile.env` to put these vars first and then all the system ones (os.environ) inside the bash subsystem, so it doesn't matter if I put first the visual in the `os.environ`. It will require some hack
